amethyst_renderer/src/renderer.rs

```diff
@@ -115,6 +119,16 @@ impl Renderer {
 
         self.device.cleanup();
 
+        // # Enforce maximum FPS
+        // Sleep if the time needed is > 10 ms for 9.5 ms
+        let time_remaining = Duration::from_secs(1) / self.max_fps - start.elapsed();
+        if time_remaining > Duration::new(0, 10000000) {
+            sleep(time_remaining * 95 / 100);
+        }
+        // Yield for the remainder of the time.
+        while start.elapsed() * self.max_fps < Duration::from_secs(1) {
```

Comment:
It is faster to store `1 sec / max_fps` in field as you always do this operation.

Reply:
You can have max fps in config yet store inverse in structure.